From 103edf8bf3216697f38dbf7b42be23ab3755cd72 Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Mon, 23 Feb 2009 13:57:38 +0800
Subject: [PATCH] ENGR00102788 Remove CONFIG_NAND_BOOT on i.MX31 3stack

1. Change NAND driver Makefile  to bulild individual nand driver on i.MX31
 and i.MX35.
2. Remove CONFIG_NAND_BOOT to common boot code which supports boot from nand
 and nor.

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 cpu/arm1136/mx31/Makefile     |    2 --
 drivers/mtd/nand/Makefile     |    2 +-
 include/configs/mx31_3stack.h |    1 -
 3 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/cpu/arm1136/mx31/Makefile b/cpu/arm1136/mx31/Makefile
index a57465f..41eb2d4 100644
--- a/cpu/arm1136/mx31/Makefile
+++ b/cpu/arm1136/mx31/Makefile
@@ -29,9 +29,7 @@ COBJS	+= generic.o
 COBJS	+= timer.o
 COBJS	+= devices.o
 
-ifdef CONFIG_NAND_BOOT
 SOBJS	= nand_load.o
-endif
 
 SRCS	:= $(SOBJS:.o=.S) $(COBJS:.o=.c)
 OBJS	:= $(addprefix $(obj),$(SOBJS) $(COBJS))
diff --git a/drivers/mtd/nand/Makefile b/drivers/mtd/nand/Makefile
index d2f9435..2adf485 100644
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -47,7 +47,7 @@ COBJS-$(CONFIG_NAND_S3C64XX) += s3c64xx.o
 COBJS-$(CONFIG_NAND_OMAP_GPMC) += omap_gpmc.o
 COBJS-$(CONFIG_NAND_PLAT) += nand_plat.o
 COBJS-$(CONFIG_MX31_NAND) += mx31_nand.o
-COBJS-y += mxc_nand.o
+COBJS-$(CONFIG_MX35) += mxc_nand.o
 endif
 
 COBJS	:= $(COBJS-y)
diff --git a/include/configs/mx31_3stack.h b/include/configs/mx31_3stack.h
index 7aaef4f..887ffe3 100644
--- a/include/configs/mx31_3stack.h
+++ b/include/configs/mx31_3stack.h
@@ -35,7 +35,6 @@
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
 
-#define CONFIG_NAND_BOOT
 #define CONFIG_SKIP_RELOCATE_UBOOT
 
 #define CONFIG_CMDLINE_TAG		1	/* enable passing of ATAGs */
-- 
1.5.4.4

